lib/config_file.rb

```diff
@@ -1,3 +1,5 @@
+require 'fileutils'
```

Comment:
Interesting. Ruby has a couple of files (notably fileutils and time/date) that have Core and Std versions, where the Core stuff works in pure mode without a require, but if you require the std library it adds extra functionality. I think the script works because I've kept to the simpler, ruby core versions of things.

That said! There's a tradeoff here between knowing that std and core behave differently and never being surprised by FileUtils working without a require, and always requiring the std version and having extra functionality bolted onto the FileUtils module. Historically I have been against increasing the surface area of modules with extra methods so that you don't have to remember what all is going to be available and/or trip you up. However, in this case I think you've hit a surprising (and delightful to me) sweet spot where this require DOES add extra functionality, but it is NOT functionality you need to be aware of or keep track of to maintain this script. AND if you don't know there's two versions of FileUtils, now you don't need to.

I've started rating all PRs and refactorings based on "does this help me or hurt me at 2am when the server is on fire and I just want to go back to bed?". This ticks the box of "we require this, so there's no surprise when we use it" without ticking the box of added complexity.

Very good call. Much appreciated!